src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,193 @@
+            } else { //проверим на совпадение по key и hash всю цепочку нод .next от curr до конца
+                Node<K,V> next;
+                while (true) { //вечный цикл (мы не знаем сколько всего нод в списке)
+                    next = curr.next;
+                    if (next == null) { //значит проверили все ноды и совпадения не выявлено
+                        curr.next = newNode(hash, key, value, null);
+                        //дописываем в конец списка новую ноду
+                        break;//и прерываем цикл
+                    }
+                    //если на предыдущих итерациях не сработал break в выше расположенном if(),
+                    //значит до конца списка еще не дошли и сечас мы проверим у next его key и hash
+                    if (next.hash == hash && (next.key == key
+                            || (key != null && key.equals(next.key)))) {
+                        existing = next;//если key и hash совпадают, то скопируем next в existing
+                        break;//и прерываем цикл
+                    }
+                    curr = next;//переходим к следующей ноде, но уже на следующей итерации
+                }
+            }
+            if (existing != null) { //если с таким key и hash уже была запись в списке
+                existing.value = value;//запишем в .value новое значение
+                return;
+            }
+        }
+        if (++size > threshold) { //увеличиваем на 1 size и после сравниваем с threshold
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K,V> e = getNode(hash(key), key);
+        return e == null ? null : e.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    static final int hash(Object key) {
+        int h;
+        return (key == null) ? 0 : (h = key.hashCode()) ^ (h >>> 16);
+    } //^ Bitwise exclusive OR (returns 0 if both bits are the same, else returns 1)
+    //>>> Unsigned Right Shift Operator (11110000 >>> 2 = 00111100)
+
+    final Node<K,V>[] resize() {
+        Node<K,V>[] oldTab = table;
+        int oldCap = (oldTab == null) ? 0 : oldTab.length;
+        int oldThr = threshold;
+        int newCap;
+        int newThr = 0;
+        if (oldCap > 0) {
+            newCap = oldCap << 1; // double capacity
+            newThr = oldThr << 1; // double threshold
+        } else { //oldCap=0,oldThr=0 (zero initial threshold signifies using defaults)
+            newCap = DEFAULT_INITIAL_CAPACITY;
+            newThr = (int)(DEFAULT_LOAD_FACTOR * newCap);
+        }
+        threshold = newThr;
+        @SuppressWarnings({"rawtypes","unchecked"})
+        Node<K,V>[] newTab = (Node<K,V>[])new Node[newCap];
+        table = newTab;
+        size = 0;
+        for (int j = 0; j < oldCap; ++j) { //нужно перенести все элементы из oldTab в newTab
+            Node<K,V> e = oldTab[j];
```

Comment:
```suggestion
            Node<K,V> curNode = oldTab[j];
```
краще інформативніший неймінг використовувати